frappe/tests/test_dashboard_connections.py

```diff
@@ -0,0 +1,285 @@
+	def tearDown(self):
+		delete_test_data()
+
+	def test_internal_link_count(self):
+		earth = frappe.get_doc(
+			{
+				"doctype": "Doctype B With Child Table With Link To Doctype A",
+				"title": "Earth",
+			}
+		)
+		earth.append(
+			"child_table",
+			{
+				"title": "Earth",
+			},
+		)
+		earth.insert()
+
+		mars = frappe.get_doc(
+			{
+				"doctype": "Doctype A With Child Table With Link To Doctype B",
+				"title": "Mars",
+			}
+		)
+		mars.append(
+			"child_table",
+			{"title": "Mars", "doctype_b_with_child_table_with_link_to_doctype_a": "Earth"},
+		)
+		mars.insert()
+
+		expected_open_count = {
+			"count": {
+				"external_links_found": [],
+				"internal_links_found": [
+					{
+						"count": 1,
+						"doctype": "Doctype B With Child Table With Link To Doctype A",
+						"names": ["Earth"],
+						"open_count": 0,
+					}
+				],
+			}
+		}
+
+		with patch.object(
+			mars.meta,
+			"get_dashboard_data",
+			return_value=get_dashboard_for_doctype_a_with_child_table_with_link_to_doctype_b(),
+		):
+			self.assertEqual(
+				get_open_count("Doctype A With Child Table With Link To Doctype B", "Mars"),
+				expected_open_count,
+			)
+
+	def test_external_link_count(self):
+		saturn = frappe.get_doc(
+			{
+				"doctype": "Doctype A With Child Table With Link To Doctype B",
+				"title": "Saturn",
+			}
+		)
+		saturn.append(
+			"child_table",
+			{
+				"title": "Saturn",
+			},
+		)
+		saturn.insert()
+
+		pluto = frappe.get_doc(
+			{
+				"doctype": "Doctype B With Child Table With Link To Doctype A",
+				"title": "Pluto",
+			}
+		)
+		pluto.append(
+			"child_table",
+			{"title": "Pluto", "doctype_a_with_child_table_with_link_to_doctype_b": "Saturn"},
+		)
+		pluto.insert()
+
+		expected_open_count = {
+			"count": {
+				"external_links_found": [
+					{"doctype": "Doctype B With Child Table With Link To Doctype A", "open_count": 0, "count": 1}
+				],
+				"internal_links_found": [],
+			}
+		}
+
+		with patch.object(
+			saturn.meta,
+			"get_dashboard_data",
+			return_value=get_dashboard_for_doctype_a_with_child_table_with_link_to_doctype_b(),
+		):
+			self.assertEqual(
+				get_open_count("Doctype A With Child Table With Link To Doctype B", "Saturn"),
+				expected_open_count,
+			)
+
+
+def create_test_data():
+	create_child_table_with_link_to_doctype_a()
+	create_child_table_with_link_to_doctype_b()
+	create_doctype_a_with_child_table_with_link_to_doctype_b()
+	create_doctype_b_with_child_table_with_link_to_doctype_a()
+	add_links_in_child_tables()
+
+
+def delete_test_data():
+	doctypes = [
+		"Child Table With Link To Doctype A",
+		"Child Table With Link To Doctype B",
+		"Doctype A With Child Table With Link To Doctype B",
+		"Doctype B With Child Table With Link To Doctype A",
+	]
+	for doctype in doctypes:
+		if frappe.db.table_exists(doctype):
+			frappe.db.delete(doctype)
+			frappe.delete_doc("DocType", doctype, force=True)
+
+
+def create_child_table_with_link_to_doctype_a():
```

Comment:
@anandbaburajan 

Use `new_doctype` utility for creating test doctypes. It automatically prefixes them with `test_.*` so it doesn't fail during coverage.

Reply:
Ah, thanks. I thought some changes in CI would be required,